### PR TITLE
[chore]: Upgrade to yarn 4

### PR DESCRIPTION
**Pull request by heshan0131**
- Upgrade default yarn version to Yarn 4.4.0 https://yarnpkg.com/blog/release/4.0
- Add .yarnrc.yml
- Added volta
- Add packageManager to package.json
- Fix github workflows
- Fix netlify build and deploy
- run yarn cache clean and yarn install and yarn bootstrap for a fresh install
- Future work might needed to fix npm publish workflow

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,7 +10,9 @@ umd/
 
 typedoc/
 
-*/**/yarn.lock
+examples/**/yarn.lock
```
Comment (ibgreen) [inline .gitignore:13]: These are the yarn4 gitignore recommendations that I have been following in the vis.gl repos https://github.com/visgl/luma.gl/blob/master/.gitignore#L33

File: website/package.json [hunk 2]
```diff
@@ -70,5 +74,6 @@
   "peerDependencies": {
     "react": "18.x",
     "react-dom": "18.x"
-  }
+  },
+  "packageManager": "yarn@4.4.0"
```
Comment (ibgreen) [inline website/package.json:78]: I had issues mixing volta and packageManager. I personally prefer volta as it makes it way easier to move between repos and directories.

Comment (ibgreen) [reply website/package.json:78]: It is possible to add a volta section that extends the base package.json, but I got some trouble when adding website to workspaces, so keep a separate volta section in website. Just my experience...

File: .gitignore (hunk 1 above)
Comment (heshan0131) [reply .gitignore:13]: I am trying different things to get netlify deployment works on this PR. So far I noticed if I don't have yarn.lock in the website and examples/demo-app folder, running yarn install in these two folders to build the website will fail with a 'issing yarn.lock' message

Comment (ibgreen) [reply .gitignore:13]: Yes you should check in the yarn.lock file

**Review by ibgreen — APPROVED** [review]

**Pull request merged**